docs/conf.py

```diff
@@ -172,8 +173,19 @@
 html_logo = "_static/logo1.png"
 
 
+def autoapi_skip_modules(app: sphinx.application.Sphinx, what: str, name: str, obj: object, skip: bool, options: dict[str, Any]) -> bool:
+    """
+    Return True if a module should be skipped in th API docs.
+    """
+    parts = name.split(".")
+    if what == "module" and any((part.startswith("_") or part == "v2") for part in parts):
```

Comment:
can we also skip [`zarr.core`](https://zarr--2002.org.readthedocs.build/en/2002/_autoapi/zarr/core/index.html)

Reply:
Instead of putting in another condition to skip `zarr.core`, could we rename it to `zarr._core` instead? This will automatically skip it in the API docs with the current logic, and also signal to folks using autocomplete in their editor that it's not public API.

Reply:
@jhamman I skipped it for now - I think it's worth renaming to zarr.core but it's a big diff so I can propose that in a follow up PR 😄